src/searcher/mod.rs

```diff
@@ -340,44 +431,36 @@ impl Searcher {
         self.config.passthru
     }
 
-    fn fill_multi_line_buffer_from_file<S: Sink>(&self, file: &File) -> Result<(), S::Error> {
+    fn fill_multi_line_buffer_from_file(&self, file: &File) -> io::Result<()> {
```

Comment:
This I just refactored to not be generic wrt a sink error type because it always returns `io::Result` and I thought I might be able to just use `io::Result` everywhere in the callback case (that ended up not being true so it also is generic wrt a `SinkError`)

(so basically things that call this just "more lazily" convert the `io::Error` -> `SinkError`)